docs/retrieval-augmented-generation/rag-repo.md

```diff
@@ -0,0 +1,110 @@
+
+### By mid-March
+
+- The capabilities in `retrieve` will be improved with more advanced functionality.  (Details TBD)
+- The InstructLab command-line interface will be updated as follows:
+  - There will be one or more commands that can be configured to index content in a vector database.  Those commands will call out to `retrieve` to do this indexing.
+  - The existing chat capability will be configurable to either use RAG (which would involve calling out to `retrieve`) or not use RAG (which would provide the current behavior).
+- Both indexing and run-time RAG will be implemented by calls that go directly from the core repo (`instructlab/instructlab`) to  `retrieve` *unless* the work to migrate SDG preprocessing to the core repo is not complete by then (in which case there may need to be calls from SDG to `retrieve` until the migration completes).
+
+### Beyond mid-March
+
+- The capabilities in `retrieve` will continue to be improved with even more advanced functionality.  (Details TBD)
+- Both indexing and run-time RAG will be implemented by calls that go directly from the core repo to `retrieve`.  The SDG repository will not directly invoke the retrieve repository since the document processing will all be done in the core repository and the outputs of that document processing will be consumed by both `retrieve` and SDG.
+
+## Alternatives
+
+- Put the indexing code in <https://github.com/instructlab/sdg> (SDG) and the run-time RAG code in <https://github.com/instructlab/instructlab> (core)
+  - Pro: This has the advantage of not adding any new dependencies.
+  - Pro: The document processing is already in SDG and chat functionality is already in core so this would require the fewest code changes.
+  - Con: Splitting the RAG functionality across multiple repositories makes it more complicated to reuse in other applications outside of InstructLab.
+  - Con: Many things we will want to do to add advanced functionality to make RAG more effective will require changes to both indexing and run-time RAG.  If those components are split across multiple repositories, that will make delivering such changes more complicated.
+- Put the indexing and run-time RAG code in <https://github.com/instructlab/instructlab> (core)
+  - Pro: This has the advantage of not adding any new dependencies.
+  - Pro: However, since the existing document processing is in SDG, the flow for indexing for RAG would be a bit complicated (i.e., it starts with a CLI call handled by the core repo then goes to SDG for some of the document processing and then back to the core for vector database indexing).   That drawback will be eliminated if/when the document processing moves into the core repository.
```

Comment:
not sure it's an issue at all that it's making a call to SDG since the `instructlab/instructlab` dependencies contain SDG anyway. We don't expect users to be using `ilab` without SDG

Reply:
however, I do agree with the below cons and don't think it's best to store any of this in `instructlab/instructlab` if we want to advertise this as 'our RAG solution'

Reply:
> not sure it's an issue at all that it's making a call to SDG since the instructlab/instructlab dependencies contain SDG anyway. We don't expect users to be using ilab without SDG

The issue is not the dependencies, it is the flow -- if document processing starts in core and then moves to SDG and then moves back to core and then moves to the RAG repo, that flow will be difficult to understand and maintain.  This should be labeled as "con" , not "pro".  I will fix that and add some more text to clarify what my concern is.